pkg/clusters/diagnostics.go

```diff
@@ -1,123 +1,17 @@
-	namespacedResources := exec.CommandContext(ctx, "kubectl", "--kubeconfig", kubeconfig.Name(), "api-resources", "--verbs=list", "--namespaced", "-o", "name") //nolint:gosec
-	namespacedResources.Stdout = &namespacedList
-	clusterResources := exec.CommandContext(ctx, "kubectl", "--kubeconfig", kubeconfig.Name(), "api-resources", "--verbs=list", "--namespaced=false", "-o", "name") //nolint:gosec
-	clusterResources.Stdout = &clusterList
-	if err := namespacedResources.Run(); err != nil {
-		return output, err
-	}
-	if err := clusterResources.Run(); err != nil {
-		return output, err
-	}
-	combinedList := strings.Split(namespacedList.String()+clusterList.String(), "\n")
-
-	for _, resource := range combinedList {
-		if resource == "" {
-			// unwanted artifact of the split
-			continue
-		}
-		var getErr bytes.Buffer
-		var descErr bytes.Buffer
-		resourceGet := exec.CommandContext(ctx, "kubectl", "--kubeconfig", kubeconfig.Name(), "get", "--show-kind", "--ignore-not-found", "-A", "-oyaml", resource) //nolint:gosec
-		resourceGet.Stdout = getAllOut
-		resourceGet.Stderr = &getErr
-		resourceDescribe := exec.CommandContext(ctx, "kubectl", "--kubeconfig", kubeconfig.Name(), "describe", "--all-namespaces", resource) //nolint:gosec
-		resourceDescribe.Stdout = describeAllOut
-		resourceDescribe.Stderr = &descErr
-		if err := resourceGet.Run(); err != nil {
-			return output, fmt.Errorf("could not get resources for cmd '%s': err %s, stderr: %s", resourceGet.String(), err, getErr.String())
-		}
-		if err := resourceDescribe.Run(); err != nil {
-			return output, fmt.Errorf("could not get resources for cmd '%s': err %s, stderr: %s", resourceDescribe.String(), err, descErr.String())
-		}
-	}
-
-	// for each Pod, run kubectl logs
-	pods, err := c.Client().CoreV1().Pods("").List(ctx, metav1.ListOptions{})
-	if err != nil {
-		return output, err
-	}
-	logsDir := filepath.Join(output, "pod_logs")
-	err = os.Mkdir(logsDir, 0o750) //nolint:gomnd
-	if err != nil {
-		return output, err
-	}
-	failedPods := make(map[string]error)
-	for _, pod := range pods.Items {
-		podLogOut, err := os.Create(filepath.Join(logsDir, fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)))
-		if err != nil {
-			failedPods[fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)] = err
-			continue
-		}
-		cmd := exec.CommandContext(ctx, "kubectl", "--kubeconfig", kubeconfig.Name(), "logs", "--all-containers", "-n", pod.Namespace, pod.Name) //nolint:gosec
-		cmd.Stdout = podLogOut
-		if err := cmd.Run(); err != nil {
-			failedPods[fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)] = err
-			continue
-		}
-		defer podLogOut.Close()
-	}
-	if len(failedPods) > 0 {
-		failedPodOut, err := os.Create(filepath.Join(output, "pod_logs_failures.txt"))
-		if err != nil {
-			return output, err
-		}
-		defer failedPodOut.Close()
-		for failed, reason := range failedPods {
-			_, err = failedPodOut.WriteString(fmt.Sprintf("%s: %v\n", failed, reason))
-			if err != nil {
-				return output, err
-			}
-		}
-	}
```

Comment:
Please consider moving this to the `gke.Cluster` implementation. Removing it will make the GKE not dump anything after the changes.

Reply:
ah of course :) I didn't realize KTF supported GKE clusters as a base that's pretty cool!